src/router_hotspot_reputation.erl

```diff
@@ -0,0 +1,232 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% == Router Hotspot Reputation ==
+%%% @end
+%%%-------------------------------------------------------------------
+-module(router_hotspot_reputation).
+
+-define(ETS, router_hotspot_reputation_ets).
+-define(OFFER_ETS, router_hotspot_reputation_offers_ets).
+-define(DEFAULT_TIMER, timer:minutes(2)).
+-define(DEFAULT_THRESHOLD, 10).
+
+%% ------------------------------------------------------------------
+%% API Exports
+%% ------------------------------------------------------------------
+-export([
+    enabled/0,
+    init/0,
+    track_offer/1,
+    track_packet/1,
+    reputations/0,
+    reputation/1,
+    denied/1,
+    reset/1,
+    crawl_offers/1
+]).
+
+%% ------------------------------------------------------------------
+%% API Functions
+%% ------------------------------------------------------------------
+
+-spec enabled() -> boolean().
+enabled() ->
+    case application:get_env(router, hotspot_reputation_enabled, false) of
+        "true" -> true;
+        true -> true;
+        _ -> false
+    end.
+
+-spec init() -> ok.
+init() ->
+    case ?MODULE:enabled() of
+        false ->
+            lager:info("router_hotspot_reputation disabled");
+        true ->
+            lager:info("router_hotspot_reputation enabled"),
+            Opts1 = [
+                public,
+                named_table,
+                set,
+                {read_concurrency, true}
+            ],
+            _ = ets:new(?ETS, Opts1),
+            Opts2 = [
+                public,
+                named_table,
+                set,
+                {write_concurrency, true}
+            ],
+            _ = ets:new(?OFFER_ETS, Opts2),
+            ok = spawn_crawl_offers(?DEFAULT_TIMER),
+            ok
+    end.
+
+-spec track_offer(Offer :: blockchain_state_channel_offer_v1:offer()) -> ok.
+track_offer(Offer) ->
+    erlang:spawn(fun() ->
+        case ?MODULE:enabled() of
```

Comment:
Rather than spawn and then test whether feature is enabled or not,
maybe test first and avoid a superfluous worker?
(but maybe there's a nuance to the flow at large that escapes me.)